### DISCUSSION TITLE
Idea: per-component Load functions

### DISCUSSION THREAD

**Discussion by isaac-mcfadyen** [sveltejs/kit#5615] [General]
### Describe the problem

- Currently, Load functions are scoped to the route, not the components.
- Sometimes this is an annoyance because a lot of components need to do their own loading/API calls/database calls etc, which requires "prop-drilling" down from the page to the component (sometimes through multiple levels).

### Describe the proposed solution

This proposal is about adding Load functions to components in addition to pages.

This would work because:
- SvelteKit already knows which components are on which page because of the `import` directives
- Load functions in components could be run in parallel because they don't depend on each other
- When the final bundle is generated, Load functions on the page and on the components could be merged into the manifest so the renderer knows that "page x requires firing x, y, and z load functions"

### Alternatives considered

- Create (or document, if it already exists) an easier way of passing around fetched data in the load function from the page to components.

### Importance

nice to have

**Comment by Conduitry** [comment]
I think this is a "no". It'd require magically injecting additional props into the instantiated child components that the parents aren't passing.

Also making this more complicated is that SSR needs to be synchronous. Each component can't be independently responsible for loading its own dependencies, because we need to render the whole page synchronously at once, and we can't wait for individual components to load what they need.

**Comment by isaac-mcfadyen** [comment]
Ah, OK. Makes sense 👍 

Is there an "intended" way of passing down props from Load functions on pages to components rather than just passing them all the way down?
